tests/test_submit_tuxbuild.py

```diff
@@ -51,102 +61,117 @@ def submit_tuxbuild(self, tuxbuild):
-        self.assertEqual("build/gcc-9-defconfig-b9979cfa-duration", metric.name)
-        self.assertEqual(541, metric.result)
+        metric = first(self.squad.metrics(name="gcc-11-defconfig-ec3ad359-duration"))
+        self.assertEqual("build/gcc-11-defconfig-ec3ad359-duration", metric.name)
+        self.assertEqual(273, metric.result)
 
+    @unittest.mock.patch.dict(os.environ, {"KERNEL_BRANCH": "master"})
     def test_submit_tuxbuild_buildset(self):
-        os.environ["KERNEL_BRANCH"] = "master"
-        proc = self.submit_tuxbuild("tests/data/submit/tuxbuild/buildset.json")
+        proc = self.submit_tuxbuild(os.path.join(self.buildset_dir, "build.json"))
         self.assertTrue(proc.ok, msg=proc.out)
         self.assertTrue(proc.err.count("Submitting 1 tests, 2 metrics") == 3)
+        project = self.squad.group("my_group").project("my_project")
 
-        build = self.squad.group("my_group").project("my_project").build("next-20201030")
+        build = project.build("next-20211225")
+        self.assertIsNotNone(build)
 
-        # Make sure metadata values match expected values
-        urls = ['https://builds.tuxbuild.com/%s/' % _id for _id in ['9NeOU1kd65bhMrL4eyI2yA', 'cjLreGasHSZj3OctZlNdpw', 'x5Mi9j6xZItTGqVtOKmnVw']]
-        configs = [url + "config" for url in urls]
-        expected_metadata = {
-            'git_repo': "https://gitlab.com/Linaro/lkft/mirrors/next/linux-next",
-            'git_ref': "master",
-            'git_sha': "4e78c578cb987725eef1cec7d11b6437109e9a49",
-            'git_short_log': '4e78c578cb98 ("Add linux-next specific files for 20201030")',
-            'git_describe': "next-20201030",
-            'kconfig': [['allnoconfig'], ['tinyconfig'], ['x86_64_defconfig']],
-            'kernel_version': "5.10.0-rc1",
-            'config': configs,
-            'download_url': urls,
-            'duration': 541,
-            'toolchain': 'gcc-8',
+        testruns = build.testruns()
+        self.assertIsNotNone(testruns)
+
+        base_metadata = {
+            "git_describe": "next-20211225",
+            "git_ref": "master",
+            "git_repo": "https://gitlab.com/Linaro/lkft/mirrors/next/linux-next",
+            "git_sha": "ea586a076e8aa606c59b66d86660590f18354b11",
+            "git_short_log": "ea586a076e8a (\"Add linux-next specific files for 20211225\")",
+            "kernel_version": "5.16.0-rc6",
+            "toolchain": "gcc-8",
         }
-        for expected_key in expected_metadata.keys():
-            self.assertEqual(expected_metadata[expected_key], getattr(build.metadata, expected_key))
 
-        # Make sure there's no extra attributes in the metadata object
-        metadata_attrs = build.metadata.__dict__
-        del metadata_attrs["id"]
-        self.assertEqual(sorted(expected_metadata.keys()), sorted(metadata_attrs.keys()))
+        expected_metadata = [
+            dict(base_metadata, **{
```

Comment:
Here's where we add the 'build_name' when we introduce that patch right?

Reply:
Yes, this is what it looks like now before changing the code.